### PR TITLE
Added distructions of thrown items into damaging blocks

### PR DESCRIPTION
**Pull request by portokaliu**
Contains:
    This is a partial fix for #1697, which only takes care of the basic need of items being destroyed when touching lava (as is specified)

How to test:
    Get an item, drop into lava, mourn its loss :)

Outstanding:
    As specified in the comments, there could be several implementations of this functionality. The others rely on items sending events when entering blocks; This would in turn be caught by this system and managed in a proper fashion if the block would be lava/has a damaging component.
    Adding a damaging component optionally to lava blocks is also an option, but first there needs to be some more exploration into the optionality of the modules as I've asked in the slack chat.

In short:
- [ ] Rely on Damaging component
- [ ] Use other system for block checking (main engine events)
- [ ] Optionally add Damaging components to blocks which deserve them

### CODE REVIEW

**Comment by Cervator** [comment]
Many torches gave up their lives to test this PR. In their honor I merge!

Yeah we could vastly improve this system, similarly to Breathing (match the blocks with creatures/items that are hurt or helped). That's another day - at least want to merge some more PRs today :-)

Thanks!
